groupby_bins tests failing with pandas 0.20

These are due to the introduction of `Interval` and `IntervalIndex` to hold the result of `pd.cut`:
http://pandas-docs.github.io/pandas-docs-travis/whatsnew.html#whatsnew-0200-enhancements-intervalindex

From the latest test of pandas 0.20 (currently in release candidate) on master:
https://travis-ci.org/pydata/xarray/jobs/224867239
```
=================================== FAILURES ===================================
_______________________ TestDataArray.test_groupby_bins ________________________
self = <xarray.tests.test_dataarray.TestDataArray testMethod=test_groupby_bins>
    def test_groupby_bins(self):
        array = DataArray(np.arange(4), dims='dim_0')
        # the first value should not be part of any group ("right" binning)
        array[0] = 99
        # bins follow conventions for pandas.cut
        # http://pandas.pydata.org/pandas-docs/stable/generated/pandas.cut.html
        bins = [0, 1.5, 5]
        bin_coords = ['(0, 1.5]', '(1.5, 5]']
        expected = DataArray([1, 5], dims='dim_0_bins',
                             coords={'dim_0_bins': bin_coords})
        # the problem with this is that it overwrites the dimensions of array!
        # actual = array.groupby('dim_0', bins=bins).sum()
        actual = array.groupby_bins('dim_0', bins).apply(lambda x: x.sum())
>       self.assertDataArrayIdentical(expected, actual)
xarray/tests/test_dataarray.py:1752: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xarray/tests/__init__.py:165: in assertDataArrayIdentical
    assert_identical(ar1, ar2)
xarray/testing.py:83: in assert_identical
    assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
a = <xarray.Dataset>
Dimensions:       (dim_0_bins: 2)
Coordinates:
  * dim_0_bins    (dim_0_bins) <U8 '(0, 1.5]' '(1.5, 5]'
Data variables:
    <this-array>  (dim_0_bins) int64 1 5
b = <xarray.Dataset>
Dimensions:       (dim_0_bins: 2)
Coordinates:
  * dim_0_bins    (dim_0_bins) object (0.0, 1.5] (1.5, 5.0]
Data variables:
    <this-array>  (dim_0_bins) int64 1 5
    def assert_identical(a, b):
        """Like :py:func:`xarray.testing.assert_equal`, but also matches the
        objects' names and attributes.
    
        Raises an AssertionError if two objects are not identical.
    
        Parameters
        ----------
        a : xarray.Dataset, xarray.DataArray or xarray.Variable
            The first object to compare.
        b : xarray.Dataset, xarray.DataArray or xarray.Variable
            The second object to compare.
    
        See also
        --------
        assert_equal, assert_allclose, Dataset.equals, DataArray.equals
        """
        import xarray as xr
        ___tracebackhide__ = True  # noqa: F841
        assert type(a) == type(b)
        if isinstance(a, xr.DataArray):
            assert a.name == b.name
            assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
        elif isinstance(a, (xr.Dataset, xr.Variable)):
>           assert a.identical(b), '{}\n{}'.format(a, b)
E           AssertionError: <xarray.Dataset>
E           Dimensions:       (dim_0_bins: 2)
E           Coordinates:
E             * dim_0_bins    (dim_0_bins) <U8 '(0, 1.5]' '(1.5, 5]'
E           Data variables:
E               <this-array>  (dim_0_bins) int64 1 5
E           <xarray.Dataset>
E           Dimensions:       (dim_0_bins: 2)
E           Coordinates:
E             * dim_0_bins    (dim_0_bins) object (0.0, 1.5] (1.5, 5.0]
E           Data variables:
E               <this-array>  (dim_0_bins) int64 1 5
xarray/testing.py:85: AssertionError
____________________ TestDataArray.test_groupby_bins_empty _____________________
self = <xarray.tests.test_dataarray.TestDataArray testMethod=test_groupby_bins_empty>
    def test_groupby_bins_empty(self):
        array = DataArray(np.arange(4), [('x', range(4))])
        # one of these bins will be empty
        bins = [0, 4, 5]
        actual = array.groupby_bins('x', bins).sum()
        expected = DataArray([6, np.nan], dims='x_bins',
                             coords={'x_bins': ['(0, 4]', '(4, 5]']})
>       self.assertDataArrayIdentical(expected, actual)
xarray/tests/test_dataarray.py:1763: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xarray/tests/__init__.py:165: in assertDataArrayIdentical
    assert_identical(ar1, ar2)
xarray/testing.py:83: in assert_identical
    assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
a = <xarray.Dataset>
Dimensions:       (x_bins: 2)
Coordinates:
  * x_bins        (x_bins) <U6 '(0, 4]' '(4, 5]'
Data variables:
    <this-array>  (x_bins) float64 6.0 nan
b = <xarray.Dataset>
Dimensions:       (x_bins: 2)
Coordinates:
  * x_bins        (x_bins) object (0, 4] (4, 5]
Data variables:
    <this-array>  (x_bins) float64 6.0 nan
    def assert_identical(a, b):
        """Like :py:func:`xarray.testing.assert_equal`, but also matches the
        objects' names and attributes.
    
        Raises an AssertionError if two objects are not identical.
    
        Parameters
        ----------
        a : xarray.Dataset, xarray.DataArray or xarray.Variable
            The first object to compare.
        b : xarray.Dataset, xarray.DataArray or xarray.Variable
            The second object to compare.
    
        See also
        --------
        assert_equal, assert_allclose, Dataset.equals, DataArray.equals
        """
        import xarray as xr
        ___tracebackhide__ = True  # noqa: F841
        assert type(a) == type(b)
        if isinstance(a, xr.DataArray):
            assert a.name == b.name
            assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
        elif isinstance(a, (xr.Dataset, xr.Variable)):
>           assert a.identical(b), '{}\n{}'.format(a, b)
E           AssertionError: <xarray.Dataset>
E           Dimensions:       (x_bins: 2)
E           Coordinates:
E             * x_bins        (x_bins) <U6 '(0, 4]' '(4, 5]'
E           Data variables:
E               <this-array>  (x_bins) float64 6.0 nan
E           <xarray.Dataset>
E           Dimensions:       (x_bins: 2)
E           Coordinates:
E             * x_bins        (x_bins) object (0, 4] (4, 5]
E           Data variables:
E               <this-array>  (x_bins) float64 6.0 nan
xarray/testing.py:85: AssertionError
___________________ TestDataArray.test_groupby_bins_multidim ___________________
self = <xarray.tests.test_dataarray.TestDataArray testMethod=test_groupby_bins_multidim>
    def test_groupby_bins_multidim(self):
        array = self.make_groupby_multidim_example_array()
        bins = [0,15,20]
        bin_coords = ['(0, 15]', '(15, 20]']
        expected = DataArray([16, 40], dims='lat_bins',
                             coords={'lat_bins': bin_coords})
        actual = array.groupby_bins('lat', bins).apply(lambda x: x.sum())
>       self.assertDataArrayIdentical(expected, actual)
xarray/tests/test_dataarray.py:1775: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xarray/tests/__init__.py:165: in assertDataArrayIdentical
    assert_identical(ar1, ar2)
xarray/testing.py:83: in assert_identical
    assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
a = <xarray.Dataset>
Dimensions:       (lat_bins: 2)
Coordinates:
  * lat_bins      (lat_bins) <U8 '(0, 15]' '(15, 20]'
Data variables:
    <this-array>  (lat_bins) int64 16 40
b = <xarray.Dataset>
Dimensions:       (lat_bins: 2)
Coordinates:
  * lat_bins      (lat_bins) object (0, 15] (15, 20]
Data variables:
    <this-array>  (lat_bins) int64 16 40
    def assert_identical(a, b):
        """Like :py:func:`xarray.testing.assert_equal`, but also matches the
        objects' names and attributes.
    
        Raises an AssertionError if two objects are not identical.
    
        Parameters
        ----------
        a : xarray.Dataset, xarray.DataArray or xarray.Variable
            The first object to compare.
        b : xarray.Dataset, xarray.DataArray or xarray.Variable
            The second object to compare.
    
        See also
        --------
        assert_equal, assert_allclose, Dataset.equals, DataArray.equals
        """
        import xarray as xr
        ___tracebackhide__ = True  # noqa: F841
        assert type(a) == type(b)
        if isinstance(a, xr.DataArray):
            assert a.name == b.name
            assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
        elif isinstance(a, (xr.Dataset, xr.Variable)):
>           assert a.identical(b), '{}\n{}'.format(a, b)
E           AssertionError: <xarray.Dataset>
E           Dimensions:       (lat_bins: 2)
E           Coordinates:
E             * lat_bins      (lat_bins) <U8 '(0, 15]' '(15, 20]'
E           Data variables:
E               <this-array>  (lat_bins) int64 16 40
E           <xarray.Dataset>
E           Dimensions:       (lat_bins: 2)
E           Coordinates:
E             * lat_bins      (lat_bins) object (0, 15] (15, 20]
E           Data variables:
E               <this-array>  (lat_bins) int64 16 40
xarray/testing.py:85: AssertionError
= 3 failed, 1771 passed, 651 skipped, 8 xfailed, 13 xpassed in 216.63 seconds ==
```

--- a/doc/whats-new.rst
+++ b/doc/whats-new.rst
@@ -28,6 +28,9 @@ Enhancements
 Bug fixes
 ~~~~~~~~~
 
+- Fix bug caused by changes to ``pandas.cut`` function (:issue:`1386`).
+By `Ryan Abernathey <https://github.com/rabernat>`_.
+
 .. _whats-new.0.9.5:
 
 v0.9.5 (17 April, 2017)

--- a/doc/whats-new.rst
+++ b/doc/whats-new.rst
@@ -28,7 +28,7 @@ Enhancements
 Bug fixes
 ~~~~~~~~~
 
-- Fix bug caused by changes to ``pandas.cut`` function (:issue:`1386`).
+- Fix test suite failure caused by changes to ``pandas.cut`` function (:issue:`1386`).
 By `Ryan Abernathey <https://github.com/rabernat>`_.
 
 .. _whats-new.0.9.5:

--- a/xarray/tests/test_dataarray.py
+++ b/xarray/tests/test_dataarray.py
@@ -1743,7 +1743,7 @@ class TestDataArray(TestCase):
         # bins follow conventions for pandas.cut
         # http://pandas.pydata.org/pandas-docs/stable/generated/pandas.cut.html
         bins = [0, 1.5, 5]
-        bin_coords = ['(0, 1.5]', '(1.5, 5]']
+        bin_coords = pd.cut(array['dim_0'], bins).categories
         expected = DataArray([1, 5], dims='dim_0_bins',
                              coords={'dim_0_bins': bin_coords})
         # the problem with this is that it overwrites the dimensions of array!
@@ -1757,9 +1757,10 @@ class TestDataArray(TestCase):
         array = DataArray(np.arange(4), [('x', range(4))])
         # one of these bins will be empty
         bins = [0, 4, 5]
+        bin_coords = pd.cut(array['x'], bins).categories
         actual = array.groupby_bins('x', bins).sum()
         expected = DataArray([6, np.nan], dims='x_bins',
-                             coords={'x_bins': ['(0, 4]', '(4, 5]']})
+                             coords={'x_bins': bin_coords})
         self.assertDataArrayIdentical(expected, actual)
         # make sure original array is unchanged
         # (was a problem in earlier versions)
@@ -1768,7 +1769,7 @@ class TestDataArray(TestCase):
     def test_groupby_bins_multidim(self):
         array = self.make_groupby_multidim_example_array()
         bins = [0,15,20]
-        bin_coords = ['(0, 15]', '(15, 20]']
+        bin_coords = pd.cut(array['lat'].values.flat, bins).categories
         expected = DataArray([16, 40], dims='lat_bins',
                              coords={'lat_bins': bin_coords})
         actual = array.groupby_bins('lat', bins).apply(lambda x: x.sum())

--- a/xarray/tests/test_dataarray.py
+++ b/xarray/tests/test_dataarray.py
@@ -1768,7 +1768,7 @@ class TestDataArray(TestCase):
 
     def test_groupby_bins_multidim(self):
         array = self.make_groupby_multidim_example_array()
-        bins = [0,15,20]
+        bins = [0, 15, 20]
         bin_coords = pd.cut(array['lat'].values.flat, bins).categories
         expected = DataArray([16, 40], dims='lat_bins',
                              coords={'lat_bins': bin_coords})

